assets/aml-benchmark/components/src/aml_benchmark/dataset_downloader/vision_dataset_adapter.py

```diff
@@ -0,0 +1,127 @@
+    def get_label(self, instance):
+        """Extract the instance's label as a string."""
+        pass
+
+    @abstractmethod
+    def get_pil_image(self, instance):
+        """Extract the instance's image as a PIL image."""
+        pass
+
+
+class Cifar10Adapter(VisionDatasetAdapter):
+    """Adapter for Cifar10 HF dataset."""
+
+    def __init__(self, dataset: Dataset):
+        """Make adapter, storing relevant information from dataset."""
+        self.label_feature = dataset.features["label"]
+
+    def get_label(self, instance):
+        """Extract the instance's label as a string."""
+        return self.label_feature.int2str(instance["label"])
+
+    def get_pil_image(self, instance):
+        """Extract the instance's image as a PIL image."""
+        return instance["img"]
+
+
+class Food101Adapter(VisionDatasetAdapter):
+    """Adapter for Food101 HF dataset."""
+
+    def __init__(self, dataset: Dataset):
+        """Make adapter, storing relevant information from dataset."""
+        self.label_feature = dataset.features["label"]
+
+    def get_label(self, instance):
+        """Extract the instance's label as a string."""
+        return self.label_feature.int2str(instance["label"])
+
+    def get_pil_image(self, instance):
+        """Extract the instance's image as a PIL image."""
+        return instance["image"]
+
+
+class PatchCamelyonAdapter(VisionDatasetAdapter):
+    """Adapter for PatchCamelyon HF dataset."""
+
+    def get_label(self, instance):
+        """Extract the instance's label as a string."""
+        return "unhealthy" if instance["label"] else "healthy"
+
+    def get_pil_image(self, instance):
+        """Extract the instance's image as a PIL image."""
+        return instance["image"]
+
+
+class Resisc45Adapter(VisionDatasetAdapter):
+    """Adapter for Resisc45 HF dataset."""
+
+    def __init__(self, dataset: Dataset):
+        """Make adapter, storing relevant information from dataset."""
+        self.label_feature = dataset.features["label"]
+
+    def get_label(self, instance):
+        """Extract the instance's label as a string."""
+        return self.label_feature.int2str(instance["label"])
+
+    def get_pil_image(self, instance):
+        """Extract the instance's image as a PIL image."""
+        return instance["image"]
+
+
+class GTSRBAdapter(VisionDatasetAdapter):
+    """Adapter for GTSRB HF dataset."""
+
+    def get_label(self, instance):
+        """Extract the instance's label as a string."""
+        # TODO(rdondera): update when dataset used in actual benchmark.
+        return str(instance["ClassId"])
+
+    def get_pil_image(self, instance):
+        """Extract the instance's image as a PIL image."""
+        return Image.open(io.BytesIO(instance["Path"]["bytes"]))
+
+
+class VisionDatasetAdapterFactory:
+    """Factory for making vision dataset adapters based on dataset names."""
+
+    @staticmethod
+    def get_adapter(dataset: Dataset) -> VisionDatasetAdapter:
```

Comment:
Since, we are retuning `None`, can we update the return type from `VisionDatasetAdapter` to `Optional[VisionDatasetAdapter]`?

Reply:
Nice catch, thank you! Done. (wondering why this didn't trigger mypy or other analyses)